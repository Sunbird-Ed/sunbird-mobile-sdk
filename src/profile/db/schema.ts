import {DbConstants} from '../../db';

export namespace ProfileEntry {

    export const _ID = '_id';
    export const TABLE_NAME = 'profiles';
    export const COLUMN_NAME_UID = 'uid';
    export const COLUMN_NAME_HANDLE = 'handle';
    export const COLUMN_NAME_AVATAR = 'avatar';
    export const COLUMN_NAME_AGE = 'age';
    export const COLUMN_NAME_GENDER = 'gender';
    export const COLUMN_NAME_STANDARD = 'standard';
    export const COLUMN_NAME_LANGUAGE = 'language';
    export const COLUMN_NAME_DAY = 'day';
    export const COLUMN_NAME_MONTH = 'month';
    export const COLUMN_NAME_IS_GROUP_USER = 'is_group_user';
    export const COLUMN_NAME_CREATED_AT = 'created_at';
    export const COLUMN_NAME_MEDIUM = 'medium';
    export const COLUMN_NAME_BOARD = 'board';
    export const COLUMN_NAME_PROFILE_IMAGE = 'profile_image';
    export const COLUMN_NAME_SUBJECT = 'subject';
    export const COLUMN_NAME_PROFILE_TYPE = 'profile_type'; // default TEACHER
    export const COLUMN_NAME_GRADE = 'grade';
    export const COLUMN_NAME_SYLLABUS = 'syllabus';
    export const COLUMN_NAME_SOURCE = 'source';
    export const COLUMN_NAME_GRADE_VALUE = 'grade_value';
    export const COLUMN_VALUE = 'value';
    export const COLUMN_NAME_CATEGORIES = 'categories';


    export interface SchemaMap {
        [COLUMN_NAME_UID]: string;
        [COLUMN_NAME_HANDLE]: string;
        [COLUMN_NAME_CREATED_AT]: number;
        [COLUMN_NAME_MEDIUM]: string;
        [COLUMN_NAME_BOARD]: string;
        [COLUMN_NAME_SUBJECT]: string;
        [COLUMN_NAME_PROFILE_TYPE]: string;
        [COLUMN_NAME_GRADE]: string;
        [COLUMN_NAME_SYLLABUS]: string;
        [COLUMN_NAME_SOURCE]: string;
        [COLUMN_NAME_GRADE_VALUE]: string;
        [COLUMN_NAME_CATEGORIES]?: string;
    }

    export const getCreateEntry: () => string = () => {
        return 'CREATE TABLE IF NOT EXISTS ' + TABLE_NAME + '(' +
            ProfileEntry._ID + DbConstants.SPACE + 'INTEGER PRIMARY KEY,' +
            COLUMN_NAME_UID + DbConstants.SPACE + DbConstants.TEXT_TYPE + ' UNIQUE NOT NULL' + DbConstants.COMMA_SEP +
            COLUMN_NAME_HANDLE + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_AVATAR + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_AGE + DbConstants.SPACE + DbConstants.INT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_GENDER + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_STANDARD + DbConstants.SPACE + DbConstants.INT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_LANGUAGE + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_DAY + DbConstants.SPACE + DbConstants.INT_TYPE + ' NOT NULL DEFAULT -1' + DbConstants.COMMA_SEP +
            COLUMN_NAME_MONTH + DbConstants.SPACE + DbConstants.INT_TYPE + ' NOT NULL DEFAULT -1' + DbConstants.COMMA_SEP +
            COLUMN_NAME_IS_GROUP_USER + DbConstants.SPACE + DbConstants.INT_TYPE + ' NOT NULL DEFAULT 0' + DbConstants.COMMA_SEP +
            COLUMN_NAME_CREATED_AT + DbConstants.SPACE + DbConstants.INT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_MEDIUM + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_BOARD + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_SUBJECT + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_PROFILE_IMAGE + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_PROFILE_TYPE + DbConstants.SPACE + DbConstants.TEXT_TYPE + ' DEFAULT "teacher"' + DbConstants.COMMA_SEP +
            COLUMN_NAME_GRADE + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_CATEGORIES + DbConstants.SPACE + DbConstants.TEXT_TYPE + '  DEFAULT \'\'' + DbConstants.COMMA_SEP +
            COLUMN_NAME_SYLLABUS + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_SOURCE + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_VALUE + DbConstants.SPACE + DbConstants.TEXT_TYPE + ' DEFAULT ""' + DbConstants.COMMA_SEP +
            COLUMN_NAME_GRADE_VALUE + DbConstants.SPACE + DbConstants.TEXT_TYPE +
            ')';
    };    export const deleteTable: (() => string) = () => {
        return 'DROP TABLE IF EXISTS ' + ProfileEntry.TABLE_NAME;

    };

    export const getAlterEntryForProfileSyllabus: (() => string) = () => {
        return 'ALTER TABLE ' + TABLE_NAME + ' ADD COLUMN ' + DbConstants.SPACE +
            ProfileEntry.COLUMN_NAME_SYLLABUS + DbConstants.TEXT_TYPE + '  DEFAULT \'\';';

    };

    export const getAlterEntryForProfileCategories: (() => string) = () => {
        return 'ALTER TABLE ' + TABLE_NAME + ' ADD COLUMN ' + DbConstants.SPACE +
            ProfileEntry.COLUMN_NAME_CATEGORIES + DbConstants.SPACE + DbConstants.TEXT_TYPE + '  DEFAULT \'\';';

    };
}

export namespace UserEntry {

    export const _ID = '_id';
    export const TABLE_NAME = 'users';
    export const COLUMN_NAME_UID = 'uid';
    export interface SchemaMap {
        [COLUMN_NAME_UID]: string;
    }

    export const getCreateEntry: (() => string) = () => {
        return 'CREATE TABLE IF NOT EXISTS ' + UserEntry.TABLE_NAME + ' (' +
            UserEntry._ID + ' INTEGER PRIMARY KEY,' +
            UserEntry.COLUMN_NAME_UID + DbConstants.SPACE + DbConstants.TEXT_TYPE +
            ' )';
    };

    export const deleteTable: (() => string) = () => {
        return 'DROP TABLE IF EXISTS ' + UserEntry.TABLE_NAME;

    };
}

export namespace GroupProfileEntry {

    export const TABLE_NAME = 'group_profile';
    export const _ID = '_id';
    export const COLUMN_NAME_UID = 'uid';
    export const COLUMN_NAME_GID = 'gid';
    export const COLUMN_NAME_EPOCH_TIMESTAMP = 'epoch_timestamp';

    export interface SchemaMap {
        [COLUMN_NAME_UID]: string;
        [COLUMN_NAME_GID]: string;
    }

    export const getCreateEntry: (() => string) = () => {
        return 'CREATE TABLE IF NOT EXISTS ' + GroupProfileEntry.TABLE_NAME + ' (' +
            GroupProfileEntry._ID + ' INTEGER PRIMARY KEY' + DbConstants.COMMA_SEP +
            COLUMN_NAME_GID + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_UID + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_EPOCH_TIMESTAMP + DbConstants.SPACE + DbConstants.INT_TYPE + DbConstants.COMMA_SEP +
            'UNIQUE (' + COLUMN_NAME_GID + DbConstants.COMMA_SEP + COLUMN_NAME_UID + ') ON CONFLICT REPLACE' +
            ')';
    };
    export const deleteTable: (() => string) = () => {
        return 'DROP TABLE IF EXISTS' + GroupProfileEntry.TABLE_NAME;
    };
}

export namespace GroupEntry {

    export const TABLE_NAME = 'groups';
    export const _ID = '_id';
    export const COLUMN_NAME_GID = 'gid';
    export const COLUMN_NAME_NAME = 'name';
    export const COLUMN_NAME_SYLLABUS = 'syllabus';
    export const COLUMN_NAME_GRADE = 'grade';
    export const COLUMN_NAME_GRADE_VALUE = 'grade_value';
    export const COLUMN_NAME_CREATED_AT = 'created_at';
    export const COLUMN_NAME_UPDATED_AT = 'updated_at';

    export interface SchemaMap {
        [COLUMN_NAME_GID]: string;
        [COLUMN_NAME_NAME]: string;
        [COLUMN_NAME_SYLLABUS]: string;
        [COLUMN_NAME_GRADE]: string;
        [COLUMN_NAME_GRADE_VALUE]: string;
        [COLUMN_NAME_CREATED_AT]: number;
        [COLUMN_NAME_UPDATED_AT]: number;
    }

    export const getCreateEntry: (() => string) = () => {
        return 'CREATE TABLE IF NOT EXISTS ' + GroupEntry.TABLE_NAME + ' (' +
            GroupEntry._ID + ' INTEGER PRIMARY KEY' + DbConstants.COMMA_SEP +
            COLUMN_NAME_GID + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_NAME + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_SYLLABUS + DbConstants.SPACE + DbConstants.TEXT_TYPE + ' DEFAULT ""' + DbConstants.COMMA_SEP +
            COLUMN_NAME_GRADE + DbConstants.SPACE + DbConstants.TEXT_TYPE + ' DEFAULT ""' + DbConstants.COMMA_SEP +
            COLUMN_NAME_GRADE_VALUE + DbConstants.SPACE + DbConstants.TEXT_TYPE + ' DEFAULT ""' + DbConstants.COMMA_SEP +
            COLUMN_NAME_CREATED_AT + ' INTEGER' + DbConstants.COMMA_SEP +
            COLUMN_NAME_UPDATED_AT + ' INTEGER' + ' )';
    };
    export const deleteTable: (() => string) = () => {
        return 'DROP TABLE IF EXISTS' + GroupEntry.TABLE_NAME;
    };

}

export namespace ImportedMetadataEntry {

    export const _ID = '_id';
    export const TABLE_NAME = 'imported_metadata';
    export const COLUMN_NAME_IMPORTED_ID = 'imported_id';
    export const COLUMN_NAME_DEVICE_ID = 'device_id';
    export const COLUMN_NAME_COUNT = 'count';

    export interface SchemaMap {
        [COLUMN_NAME_IMPORTED_ID]: string;
        [COLUMN_NAME_DEVICE_ID]: string;
        [COLUMN_NAME_COUNT]: string;
    }
    export const getCreateEntry: (() => string) = () => {
        return 'CREATE TABLE IF NOT EXISTS ' + ImportedMetadataEntry.TABLE_NAME + ' (' +
            ImportedMetadataEntry._ID + ' INTEGER PRIMARY KEY,' +
            ImportedMetadataEntry.COLUMN_NAME_IMPORTED_ID + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            ImportedMetadataEntry.COLUMN_NAME_DEVICE_ID + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            ImportedMetadataEntry.COLUMN_NAME_COUNT + DbConstants.SPACE + DbConstants.INT_TYPE +
            ' )';
    };

    export const getDeleteEntry: (() => string) = () => {
        return 'DROP TABLE IF EXISTS ' + ImportedMetadataEntry.TABLE_NAME;
    };

}

export namespace LearnerAssessmentsEntry {

    export const _ID = '_id';
    export const TABLE_NAME = 'learner_assessments';
    export const COLUMN_NAME_UID = 'uid';
    export const COLUMN_NAME_CONTENT_ID = 'content_id';
    export const COLUMN_NAME_QID = 'qid';
    export const COLUMN_NAME_Q_INDEX = 'qindex';
    export const COLUMN_NAME_CORRECT = 'correct';
    export const COLUMN_NAME_SCORE = 'score';
    export const COLUMN_NAME_MAX_SCORE = 'max_score';
    export const COLUMN_NAME_TIME_SPENT = 'time_spent';
    export const COLUMN_NAME_RES = 'res';
    export const COLUMN_NAME_TIMESTAMP = 'timestamp';
    export const COLUMN_NAME_Q_DESC = 'qdesc';
    export const COLUMN_NAME_Q_TITLE = 'qtitle';
    export const COLUMN_NAME_HIERARCHY_DATA = 'h_data';
    export const COLUMN_NAME_TOTAL_TS = 'total_ts';
    export const COLUMN_NAME_MARKS = 'marks';
    export const COLUMN_NAME_COUNT = 'occurence_count';
    export const COLUMN_NAME_TOTAL_MAX_SCORE = 'sum_max_score';
    export const COLUMN_NAME_USERS_COUNT = 'users_count';
    export const COLUMN_NAME_HANDLE = 'handle';



    export interface SchemaMap {
        [COLUMN_NAME_UID]: string;
        [COLUMN_NAME_CONTENT_ID]: string;
        [COLUMN_NAME_QID]: string;
        [COLUMN_NAME_Q_INDEX]: number;
        [COLUMN_NAME_CORRECT]: number;
        [COLUMN_NAME_SCORE]: number;
        [COLUMN_NAME_MAX_SCORE]: number;
        [COLUMN_NAME_TIME_SPENT]: number;
        [COLUMN_NAME_RES]: string;
        [COLUMN_NAME_TIMESTAMP]: number;
        [COLUMN_NAME_Q_DESC]: string;
        [COLUMN_NAME_Q_TITLE]: string;
        [COLUMN_NAME_HIERARCHY_DATA]: string;
    }

    export interface QuestionReportsSchema extends SchemaMap {
        [COLUMN_NAME_MARKS]: number;
        [COLUMN_NAME_COUNT]: number;
        [COLUMN_NAME_TOTAL_MAX_SCORE]: number;
    }

    export interface AccuracySchema {
        [COLUMN_NAME_QID]: string;
        [COLUMN_NAME_USERS_COUNT]: number;
    }

    export interface UserReportSchema {
        [COLUMN_NAME_TOTAL_TS]: number;
        [COLUMN_NAME_SCORE]: number;
        [COLUMN_NAME_HIERARCHY_DATA]: string;
        [COLUMN_NAME_CONTENT_ID]: string;
        [COLUMN_NAME_UID]: string;
        [COLUMN_NAME_HANDLE]: string;
        [COLUMN_NAME_TIME_SPENT]: number;
    }

    export const getCreateEntry: (() => string) = () => {
        return 'CREATE TABLE IF NOT EXISTS ' + TABLE_NAME + ' (' +
            _ID + ' INTEGER PRIMARY KEY,' +
            COLUMN_NAME_UID + DbConstants.SPACE + DbConstants.TEXT_TYPE + ' NOT NULL' + DbConstants.COMMA_SEP +
            COLUMN_NAME_CONTENT_ID + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_QID + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_Q_INDEX + DbConstants.SPACE + DbConstants.INT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_CORRECT + DbConstants.SPACE + DbConstants.INT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_SCORE + DbConstants.SPACE + DbConstants.INT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_MAX_SCORE + DbConstants.SPACE + DbConstants.INT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_TIME_SPENT + DbConstants.SPACE + DbConstants.INT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_RES + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_TIMESTAMP + DbConstants.SPACE + DbConstants.INT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_Q_DESC + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_Q_TITLE + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_HIERARCHY_DATA + DbConstants.SPACE + DbConstants.TEXT_TYPE + '  DEFAULT \'\' ' +
            ' )';
    };

    export const getDeleteEntry: (() => string) = () => {
        return 'DROP TABLE IF EXISTS ' + TABLE_NAME;
    };

}

export namespace LearnerSummaryEntry {

    export const _ID = '_id';
    export const TABLE_NAME = 'learner_content_summary';
    export const COLUMN_NAME_UID = 'uid';
    export const COLUMN_NAME_CONTENT_ID = 'content_id';
    export const COLUMN_NAME_AVG_TS = 'avg_ts';
    export const COLUMN_NAME_SESSIONS = 'sessions';
    export const COLUMN_NAME_TOTAL_TS = 'total_ts';
    export const COLUMN_NAME_LAST_UPDATED_ON = 'last_updated_on';
    export const COLUMN_NAME_HIERARCHY_DATA = 'h_data';
    export const COLUMN_NAME_NO_OF_QUESTIONS = 'no_of_questions';
    export const COLUMN_NAME_CORRECT_ANSWERS = 'correct_answers';
    export const COLUMN_NAME_TOTAL_TIME_SPENT = 'total_time_spent';
    export const COLUMN_NAME_TOTAL_MAX_SCORE = 'total_max_score';
    export const COLUMN_NAME_TOTAL_SCORE = 'total_score';


    export interface SchemaMap {
        [COLUMN_NAME_UID]: string;
        [COLUMN_NAME_CONTENT_ID]: string;
        [COLUMN_NAME_AVG_TS]: number;
        [COLUMN_NAME_SESSIONS]?: number;
        [COLUMN_NAME_TOTAL_TS]: number;
        [COLUMN_NAME_LAST_UPDATED_ON]: number;
        [COLUMN_NAME_HIERARCHY_DATA]: string;
        [COLUMN_NAME_NO_OF_QUESTIONS]?: number;
        [COLUMN_NAME_CORRECT_ANSWERS]?: number;
        [COLUMN_NAME_TOTAL_TIME_SPENT]?: number;
        [COLUMN_NAME_TOTAL_MAX_SCORE]?: number;
        [COLUMN_NAME_TOTAL_SCORE]?: number;
    }


    export const getCreateEntry: (() => string) = () => {
        return 'CREATE TABLE IF NOT EXISTS ' + TABLE_NAME + ' (' +
            _ID + ' INTEGER PRIMARY KEY,' +
            COLUMN_NAME_UID + DbConstants.SPACE + DbConstants.TEXT_TYPE + ' NOT NULL' + DbConstants.COMMA_SEP +
            COLUMN_NAME_CONTENT_ID + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_AVG_TS + DbConstants.SPACE + DbConstants.REAL_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_SESSIONS + DbConstants.SPACE + DbConstants.INT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_TOTAL_TS + DbConstants.SPACE + DbConstants.INT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_LAST_UPDATED_ON + DbConstants.SPACE + DbConstants.SPACE + DbConstants.INT_TYPE + DbConstants.COMMA_SEP +
            COLUMN_NAME_HIERARCHY_DATA + DbConstants.SPACE + DbConstants.TEXT_TYPE + DbConstants.COMMA_SEP +
            ' UNIQUE (' + COLUMN_NAME_UID + DbConstants.SPACE + DbConstants.COMMA_SEP + COLUMN_NAME_CONTENT_ID
            + DbConstants.COMMA_SEP + COLUMN_NAME_HIERARCHY_DATA + ') ON CONFLICT REPLACE' +
            ' )';
    };

    export const getDeleteEntry: (() => string) = () => {
        return 'DROP TABLE IF EXISTS ' + TABLE_NAME;
    };

}


