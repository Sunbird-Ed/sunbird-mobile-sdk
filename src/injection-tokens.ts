export const CsInjectionTokens = {
    HTTP_SERVICE: Symbol.for('CS_HTTP_SERVICE'),
    GROUP_SERVICE: Symbol.for('CS_GROUP_SERVICE'),
    COURSE_SERVICE: Symbol.for('CS_COURSE_SERVICE'),
    USER_SERVICE: Symbol.for('CS_USER_SERVICE'),
    DISCUSSION_SERVICE: Symbol.for('CS_DISCUSSION_SERVICE'),
    CONTENT_SERVICE: Symbol.for('CS_CONTENT_SERVICE'),
    NOTIFICATION_SERVICE_V2: Symbol.for('CS_NOTIFICATION_SERVICE'),
    CERTIFICATE_SERVICE: Symbol.for('CS_CERTIFICATE_SERVICE'),
    FRAMEWORK_SERVICE: Symbol.for('CS_FRAMEWORK_SERVICE'),
};

export const InjectionTokens = {
    CONTAINER: Symbol.for('CONTAINER'),
    DB_VERSION: Symbol.for('DB_VERSION'),
    DB_MIGRATION_LIST: Symbol.for('DB_MIGRATION_LIST'),
    SDK_CONFIG: Symbol.for('SDK_CONFIG'),
    DEVICE_INFO: Symbol.for('DEVICE_INFO'),
    SHARED_PREFERENCES: Symbol.for('SHARED_PREFERENCES'),
    EVENTS_BUS_SERVICE: Symbol.for('EVENTS_BUS_SERVICE'),
    DB_SERVICE: Symbol.for('DB_SERVICE'),
    APP_INFO: Symbol.for('APP_INFO'),
    API_SERVICE: Symbol.for('API_SERVICE'),
    AUTH_SERVICE: Symbol.for('AUTH_SERVICE'),
    FILE_SERVICE: Symbol.for('FILE_SERVICE'),
    SYSTEM_SETTINGS_SERVICE: Symbol.for('SYSTEM_SETTINGS_SERVICE'),
    FRAMEWORK_SERVICE: Symbol.for('FRAMEWORK_SERVICE'),
    PROFILE_SERVICE: Symbol.for('PROFILE_SERVICE'),
    GROUP_SERVICE: Symbol.for('GROUP_SERVICE'),
    GROUP_SERVICE_DEPRECATED: Symbol.for('GROUP_SERVICE_DEPRECATED'),
    ZIP_SERVICE: Symbol.for('ZIP_SERVICE'),
    TELEMETRY_SERVICE: Symbol.for('TELEMETRY_SERVICE'),
    CONTENT_FEEDBACK_SERVICE: Symbol.for('CONTENT_FEEDBACK_SERVICE'),
    FORM_SERVICE: Symbol.for('FORM_SERVICE'),
    PAGE_ASSEMBLE_SERVICE: Symbol.for('PAGE_ASSEMBLE_SERVICE'),
    FRAMEWORK_UTIL_SERVICE: Symbol.for('FRAMEWORK_UTIL_SERVICE'),
    DOWNLOAD_SERVICE: Symbol.for('DOWNLOAD_SERVICE'),
    CONTENT_SERVICE: Symbol.for('CONTENT_SERVICE'),
    COURSE_SERVICE: Symbol.for('COURSE_SERVICE'),
    SUMMARIZER_SERVICE: Symbol.for('SUMMARIZER_SERVICE'),
    PLAYER_SERVICE: Symbol.for('PLAYER_SERVICE'),
    CACHED_ITEM_STORE: Symbol.for('CACHED_ITEM_STORE'),
    KEY_VALUE_STORE: Symbol.for('KEY_VALUE_STORE'),
    TELEMETRY_DECORATOR: Symbol.for('TELEMETRY_DECORATOR'),
    STORAGE_SERVICE: Symbol.for('STORAGE_SERVICE'),
    NOTIFICATION_SERVICE: Symbol.for('NOTIFICATION_SERVICE'),
    ERROR_LOGGER_SERVICE: Symbol.for('ERROR_LOGGER_SERVICE'),
    NETWORKINFO_SERVICE: Symbol.for('NETWORKINFO_SERVICE'),
    SEARCH_HISTORY_SERVICE: Symbol.for('SEARCH_HISTORY_SERVICE'),
    CODEPUSH_EXPERIMENT_SERVICE: Symbol.for('CODEPUSH_EXPERIMENT_SERVICE'),
    FAQ_SERVICE: Symbol.for('FAQ_SERVICE'),
    DEVICE_REGISTER_SERVICE: Symbol.for('DEVICE_REGISTER_SERVICE'),
    CONTENT_RATING_SERVICE: Symbol.for('CONTENT_RATING_SERVICE'),
    ARCHIVE_SERVICE: Symbol.for('ARCHIVE_SERVICE'),
    NETWORK_QUEUE: Symbol.for('NETWORK_QUEUE'),
    CLASS_ROOM_SERVICE: Symbol.for('CLASS_ROOM_SERVICE'),
    DISCUSSION_SERVICE: Symbol.for('DISCUSSION_SERVICE'),
    SEGMENTATION_SERVICE: Symbol.for('SEGMENTATION_SERVICE'),
    DEBUGGING_SERVICE: Symbol.for('DEBUGGING_SERVICE'),
    NOTIFICATION_SERVICE_V2: Symbol.for('NOTIFICATION_SERVICE_V2'),
    CERTIFICATE_SERVICE: Symbol.for('CERTIFICATE_SERVICE'),
};
