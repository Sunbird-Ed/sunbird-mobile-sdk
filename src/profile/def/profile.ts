import {ServerProfile} from './server-profile';

export {
    UserDeclaration, UserFeedEntry, UserFeedCategory, UserFeedStatus
} from '@project-sunbird/client-services/models/user';

export enum ProfileType {
    STUDENT = 'student',
    TEACHER = 'teacher',
    ADMIN = 'administrator',
    PARENT = 'parent',
    OTHER = 'other',
    NONE = 'none'
}

export enum ProfileSource {
    SERVER = 'server',
    LOCAL = 'local'
}

export interface Profile {
    uid: string;
    handle: string;
    createdAt?: number;
    medium?: string[];
    board?: string[];
    subject?: string[];
    profileType: ProfileType;
    grade?: string[];
    syllabus?: string[];
    source: ProfileSource;
    gradeValue?: { [key: string]: any };
    serverProfile?: ServerProfile;
    categories?: string;
}

