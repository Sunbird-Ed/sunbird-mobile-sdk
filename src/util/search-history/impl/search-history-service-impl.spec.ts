import {Container} from 'inversify';
import {InjectionTokens} from '../../../injection-tokens';
import {SearchHistoryServiceImpl} from './search-history-service-impl';
import {SearchHistoryService} from '..';
import {DbService} from '../../../db';
import {ProfileService, ProfileSession} from '../../../profile';
import {Observable} from 'rxjs';
import {SearchHistoryEntry} from '../db/schema';

describe('SearchHistoryServiceImpl', () => {
    let searchHistoryService: SearchHistoryService;

    const container = new Container();
    const dbServiceMock: Partial<DbService> = {};
    const profileServiceMock: Partial<ProfileService> = {};

    beforeAll(() => {
        container.bind<SearchHistoryService>(InjectionTokens.SEARCH_HISTORY_SERVICE).to(SearchHistoryServiceImpl);
        container.bind<ProfileService>(InjectionTokens.PROFILE_SERVICE).toConstantValue(profileServiceMock as ProfileService);
        container.bind<DbService>(InjectionTokens.DB_SERVICE).toConstantValue(dbServiceMock as DbService);

        searchHistoryService = container.get<SearchHistoryService>(InjectionTokens.SEARCH_HISTORY_SERVICE);
    });

    it('should return an instance of SearchHistoryServiceImpl from container', () => {
        // assert
        expect(searchHistoryService).toBeTruthy();
    });

    it('should add entry to db for current profile on addEntry()', (done) => {
        // arrange
        const mockSession: ProfileSession = new ProfileSession('SAMPLE_UID');
        profileServiceMock.getActiveProfileSession = jest.fn(() => Observable.of(mockSession));
        dbServiceMock.insert = jest.fn(() => Observable.of(undefined));
        dbServiceMock.execute = jest.fn(() => Observable.of(undefined));

        // act
        searchHistoryService.addEntry({query: 'SAMPLE_QUERY', namespace: 'SAMPLE_NAMESPACE'})
            .subscribe(() => {
                // assert
                expect(profileServiceMock.getActiveProfileSession).toBeCalled();
                expect(dbServiceMock.insert).toBeCalledWith(expect.objectContaining({
                    modelJson: expect.objectContaining({
                        [SearchHistoryEntry.COLUMN_NAME_QUERY]: 'SAMPLE_QUERY',
                        [SearchHistoryEntry.COLUMN_NAME_NAMESPACE]: 'SAMPLE_NAMESPACE',
                        [SearchHistoryEntry.COLUMN_NAME_USER_ID]: 'SAMPLE_UID'
                    })
                }));
                expect(dbServiceMock.execute).toBeCalled();
                done();
            });
    });

    it('should return entries for current Profile on getEntries()', (done) => {
        // arrange
        const mockSession: ProfileSession = new ProfileSession('SAMPLE_UID');
        profileServiceMock.getActiveProfileSession = jest.fn(() => Observable.of(mockSession));
        dbServiceMock.execute = jest.fn(() => Observable.of(<SearchHistoryEntry.SchemaMap[]>[
            {
                [SearchHistoryEntry._ID]: '1',
                [SearchHistoryEntry.COLUMN_NAME_USER_ID]: 'SAMPLE_UID',
                [SearchHistoryEntry.COLUMN_NAME_QUERY]: 'SAMPLE_QUERY',
                [SearchHistoryEntry.COLUMN_NAME_TIME_STAMP]: 1,
                [SearchHistoryEntry.COLUMN_NAME_NAMESPACE]: 'SAMPLE_NAMESPACE'
            },
            {
                [SearchHistoryEntry._ID]: '2',
                [SearchHistoryEntry.COLUMN_NAME_USER_ID]: 'SAMPLE_UID_2',
                [SearchHistoryEntry.COLUMN_NAME_QUERY]: 'SAMPLE_QUERY_2',
                [SearchHistoryEntry.COLUMN_NAME_TIME_STAMP]: 2,
                [SearchHistoryEntry.COLUMN_NAME_NAMESPACE]: 'SAMPLE_NAMESPACE_2'
            }
        ]));

        // act
        searchHistoryService.getEntries({namespace: 'SAMPLE_NAMESPACE', limit: 10})
            .subscribe((results) => {
                // assert
                expect(results).toMatchSnapshot();
                expect(profileServiceMock.getActiveProfileSession).toBeCalled();
                expect(dbServiceMock.execute).toBeCalled();

                done();
            });
    });
});
