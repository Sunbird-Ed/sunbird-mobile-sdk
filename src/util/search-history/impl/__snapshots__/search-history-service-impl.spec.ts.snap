// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHistoryServiceImpl should return entries for current Profile on getEntries() 1`] = `
Array [
  Object {
    "query": "SAMPLE_QUERY",
    "timestamp": 1,
    "uid": "SAMPLE_UID",
  },
  Object {
    "query": "SAMPLE_QUERY_2",
    "timestamp": 2,
    "uid": "SAMPLE_UID_2",
  },
]
`;
